src/plugins/simulator/media/led_medium.cpp

```diff
@@ -38,9 +38,9 @@ namespace argos {
          if(strPosIndexMethod == "grid") {
             size_t punGridSize[3];
             if(!NodeAttributeExists(t_tree, "grid_size")) {
-               punGridSize[0] = cArenaSize.GetX();
-               punGridSize[1] = cArenaSize.GetY();
-               punGridSize[2] = cArenaSize.GetZ();
+               punGridSize[0] = static_cast<UInt32>(cArenaSize.GetX());
```

Comment:
Are these casts really necessary? Is it a warning that makes them necessary?

Reply:
Same thing here